### ISSUE TITLE
MinkLoc3dV2 Mismatch issue

### ISSUE THREAD

**Issue by uljad** [csiro-robotics/Wild-Places#5]
When testing the checkpoint of MinkLoc3dV2 (loaded to the tuned architecture with the required parameter additions) on the Karawatha pickles generated with your script, **I get no matches** . 

I think the problem is that the pickles generated with your code do not contain keys to get the true neighbors like with the Oxford dataset

**Comment by uljad** [comment]
There are some dimesional differences in the recall calculation loops for MinkLoc3Dv2 [evaluation code](https://github.com/jac99/MinkLoc3Dv2/tree/main/eval) which if corrected make everything work as intended in the paper(s).
The only inconvenience is that it takes longer to evaluate because there are many more poses if I am doing things correctly

**Comment by JBKnights** [comment]
Hi, would you be able to elaborate on what you mean by dimensional differences?  We're currently working on providing clearer training / evaluation instructions for the approaches we benchmark in the paper, some clarification on the problem you're facing here and your implemented solution would be very helpful!

**Comment by uljad** [comment]
Hello,
My problem is that I do not get the same performance with the database embeddings. I am using the weights from the pth files in the WildPlaces website and loading them to the network with some adjustments. Do you mind sharing your adjustments to the original Minkloc3d_v2?


Here are my adjustments:

```
[MODEL]
model=MinkLoc
planes=64,128,128,128,64
layers=1,1,1,1,1
num_top_down=2
conv0_kernel_size=5
feature_size=256
block=ECABasicBlock
pooling=GeM

coordinates=cartesian
quantization_step=0.01

normalize_embeddings=False

```

**Comment by JBKnights** [comment]
Hi, those adjustments seem correct for the most part - the only major difference is we use quantization_step=0.8 , not 0.01.  Does this change bring you closer to our reported results?

**Comment by kavisha725** [comment]
Closing due to inactivity. Feel free to reopen if needed.

**Comment by HaedamOh** [comment]
Hi, I am trying to reproduce the same result from WildPlace for MinkLoc3d. What clouds did you use in WildPlace for evaluation? Clouds or Clouds_downsampled ?

**Comment by HaedamOh** [comment]
Hello, I have found the recall / F1 score of MinkLoc3d pre-trained model evaluated on WildPlace K-04 is very low (about 40% recall). Here I used the same setting as above and just used your pretrained model to evaluate. 
Is there any pre-processing has been done in MinkLoc3d? or did I miss something? (Could you also please check pretrained model for MinkLoc3d?) 

![image](https://github.com/csiro-robotics/Wild-Places/assets/109588147/ba60b6db-22a3-45a5-9e4f-118f9c579acc)

**Comment by shenyanqing1105** [comment]
> Hello, I have found the recall / F1 score of MinkLoc3d pre-trained model evaluated on WildPlace K-04 is very low (about 40% recall). Here I used the same setting as above and just used your pretrained model to evaluate. Is there any pre-processing has been done in MinkLoc3d? or did I miss something? (Could you also please check pretrained model for MinkLoc3d?)
> 
> ![image](https://private-user-images.githubusercontent.com/109588147/288955558-ba60b6db-22a3-45a5-9e4f-118f9c579acc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyMzAxNzQsIm5iZiI6MTcxMDIyOTg3NCwicGF0aCI6Ii8xMDk1ODgxNDcvMjg4OTU1NTU4LWJhNjBiNmRiLTIyYTMtNDVhNS05ZTRmLTExOGY5YzU3OWFjYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxMlQwNzUxMTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YzhjNjYwNGFjYmQwZTNjOTRjYWU1Y2EyNWViODhjMTQwMGQ5ODQ4N2Q3ZGY2NjgzNWRlMjhlMGVlYTU5Y2NjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.sbscClx8IqX-8Sr48P7J_WNm8B3qVfnMeaLKhQx4T8M)

Hi, have you solved this problem? I just got the same one.

**Comment by shenyanqing1105** [comment]
> Hello, My problem is that I do not get the same performance with the database embeddings. I am using the weights from the pth files in the WildPlaces website and loading them to the network with some adjustments. Do you mind sharing your adjustments to the original Minkloc3d_v2?
> 
> Here are my adjustments:
> 
> ```
> [MODEL]
> model=MinkLoc
> planes=64,128,128,128,64
> layers=1,1,1,1,1
> num_top_down=2
> conv0_kernel_size=5
> feature_size=256
> block=ECABasicBlock
> pooling=GeM
> 
> coordinates=cartesian
> quantization_step=0.01
> 
> normalize_embeddings=False
> ```

Hi, have you solved this problem? I just got the same one.

**Comment by mramezani64** [comment]
@shenyanqing1105 @HaedamOh Have you tried changing Cartesian to `polar` coordinates with `quantization_step=0.8, 0.15, 0.15` to see how it goes?

**Comment by shenyanqing1105** [comment]
> @shenyanqing1105 @HaedamOh Have you tried changing Cartesian to `polar` coordinates with `quantization_step=0.8, 0.15, 0.15` to see how it goes?

Thanks for your suggestion! But it still doesn't work.

MinkLoc3Dv2 model definition
[MODEL]
model=MinkLoc
planes=64,128,128,128,64
layers=1,1,1,1,1
num_top_down=2
conv0_kernel_size=5
feature_size=256
block=ECABasicBlock
pooling=GeM

coordinates= polar
quantization_step=0.15, 0.15, 0.8

normalize_embeddings=False

**Comment by shenyanqing1105** [comment]
> > @shenyanqing1105 @HaedamOh Have you tried changing Cartesian to `polar` coordinates with `quantization_step=0.8, 0.15, 0.15` to see how it goes?
> 
> Thanks for your suggestion! But it still doesn't work.
> 
> MinkLoc3Dv2 model definition [MODEL] model=MinkLoc planes=64,128,128,128,64 layers=1,1,1,1,1 num_top_down=2 conv0_kernel_size=5 feature_size=256 block=ECABasicBlock pooling=GeM
> 
> coordinates= polar quantization_step=0.15, 0.15, 0.8
> 
> normalize_embeddings=False

Ah! Now I finally did it! Also change **normalize_embeddings** to True.

Thanks!
